birdhouse/components/weaver/docker-compose-extra.yml

```diff
@@ -0,0 +1,124 @@
+version: "3.4"
+
+x-logging:
+  &default-logging
+  driver: "json-file"
+  options:
+    max-size: "200k"
+    max-file: "10"
+
+services:
+
+  proxy:
+    volumes:
+      - ./components/weaver/conf.extra-service.d:/etc/nginx/conf.extra-service.d/weaver
+    links:
+      - weaver
+
+  magpie:
+    volumes:
+      - ./components/weaver/config/magpie/config.yml:/opt/local/src/magpie/config/permissions/weaver-permissions.cfg:ro
+      - ./components/weaver/config/magpie/config.yml:/opt/local/src/magpie/config/providers/weaver-provider.cfg:ro
+
+  # FIXME:
+  #   Weaver API shouldn't be one doing docker run (jobs), but it is actually the one
+  #   running the cwltool app from the received WPS Execute request sent by the worker.
+  #   Configs must be moved to the 'worker' with implementation of direct job monitoring.
+  #   (https://github.com/crim-ca/weaver/issues/21)
+  weaver:
+    image: pavics/weaver:${WEAVER_VERSION}-manager
+    container_name: ${WEAVER_MANAGER_NAME}
+    ports:
+      - "4001:4001"
+    environment:
+      HOSTNAME: ${PAVICS_FQDN}
+      FORWARDED_ALLOW_IPS: "*"
+      DOCKER_HOST: "tcp://docker-proxy:2375"   # FIXME: see above docker-sibling note
+    #env_file:
+    #  - ./config/mongodb/credentials.env
+    # ensure other referenced services are created beforehand to avoid booting timing issues
+    depends_on:
+      - catalog
+      - finch
+      - flyingpigeon
+      - hummingbird
+      - malleefowl
+      - raven
+      - worker
+    links:
+      - mongodb
+    volumes:
+      - ./components/weaver/config/weaver/weaver.ini:/opt/local/src/weaver/config/weaver.ini
+      - ./components/weaver/config/weaver/data_sources.json:/opt/local/src/weaver/config/data_sources.json
+      - ./components/weaver/config/weaver/request_options.yml:/opt/local/src/weaver/config/request_options.yml
+      - ./components/weaver/config/weaver/wps_processes.yml:/opt/local/src/weaver/config/wps_processes.yml
+      - /usr/bin/docker:/usr/bin/docker   # FIXME: see above docker-sibling note
+      - ~/.docker/config.json:/root/.docker/config.json   # FIXME: see above docker-sibling note
```

Comment:
Whooo, what do you need special from the user's docker config?  Is it documented somewhere?  Not sure I like this either.

Reply:
It is needed for access some dockers. If not provided, remote private repositories cannot be accessed. 
We can make it a separate extra component. Not a 'hard must' and not needed if every reference dockers is public. 

Feature to provide auth via API is not implemented in Weaver (at least for now), and I'm not sure if CWL even supports it. 
That feature was pushed aside as not a priority.

Reply:
Hum, I agree with @tlvu, Weaver only needs access to the Application Packages repositories. Those are not the same as the PAVICS components repositories. I would change that line with `./components/weaver/config/docker/config.json:/root/.docker/config.json`